source/intel.js

```diff
@@ -100,10 +100,50 @@ const display_tech_trading = () => {
 	}
 	return tech_trade_screen
 }
+let cached_events = [];
+
+const update_event_cache = (fetchSize) => {
+		jQuery.ajax({
+			type: 'POST',
+			url: "/trequest/fetch_game_messages",
+			async: true,
+			data: {
+				type: "fetch_game_messages",
+				count: cached_events.length > 0 ? fetchSize : 100000,
```

Comment:
I ask this because I set `100,000` as an arbitrarily high number. I have one game though that has `3,797` ticks so I will test to see if in that game I have hit the event limit.

Reply:
I am going to test `-1` and `0` to see if the server returns unlimited. Otherwise we do not need to limit `fetchSize`.

Reply:
Note actually fetching too many messages `100,000` when there exists a lot of messages returns a error 500.

Reply:
The actual max size the server will accept is `Math.pow(2,31)-1` through trial an error.

Although, I keep getting ERR 500 when asking for more messages then there actually is. Maybe there is an endpoint to get the number of events that exist.

Reply:
There is no need for this limit, I could code it without it. I did it this way because the original code did it this way, I didn't choose the limit or ask whether it was tested.

Depending on philosophy, you could say "well this is as it was, therefore there is no new bug here, i'll take this and we can fix it in a follow up PR" or you can say "let's fix it by not having the limit".

To remove the limit, I would start by fetching k*2^n messages repeatedly until a fetch failed to find new messages. so the scheme would be:
fetch #1: fetch 10
fetch #2: fetch 10*2 @ offset 10
fetch #3: fetch 10*2*2 @ offset 10*2
and so on.

Eventually one of those fetches will not find more events. Then you have the base cache loaded. 

This is however an entirely new approach rather than just using a "big limit".

Another possibility to fix it is to use a "small but usually plenty big enough limit" for the first fetch. Say for example 30000; most games don't have that many messages. If the response comes back and is 30,000 strong, populate the base cache but then run the existing code that tops it up. A problem with this approach is that the special case of more than 30,000 messages will be rare, so bugs if any will be rare and hard to detect.

So a final very simple approach is the same as that but with a base fetch of just 1000 messages. That'll be fast when the game is young, but the 'top up' code will be triggered regularly enough that we can be sure it works.

So several options to choose from.